apis/Google.Cloud.Firestore/Google.Cloud.Firestore/DeletedFieldAttribute.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright 2018, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace Google.Cloud.Firestore
+{
+    // TODO: Do we really want this? More present for consistency than anything else.
```

Comment:
I like the consistency of including this. But if it has *absolutely no use whatsoever* then it should probably be removed, to remove the possibility of confusion.

Reply:
We'll see what the Firestore team says.

Reply:
Yeah, we left it out of Android / Java because it doesn't make much sense.  In general we don't expect apps to use FieldValue.Delete as part of their normal operation because their objects likely match a "schema" which has a consistent set of fields, and even for "optional" fields you generally want to store `null` rather than omit the field, so you can still query/sort on it.  And if you're using object-mapping then it's even clearer that you have a schema with a well-defined set of properties.

I can kinda' conceive of one use for it in object-mapping as a form of schema migration (if you are writing your objects with SetOptions.MergeAll and at some point decide to remove some field you used to use, you could change your object to have the DeletedField attribute on that field to delete it whenever writing objects), but I'm not sure this is realistic and I think I'd rather wait and see if any user asks for it.

So I'd be inclined to remove it for now to keep our surface area minimal and consistent with the Java SDK and revisit if we ever get user feedback asking for it.  I don't feel super strongly though if you disagree.